src/modules/price-router/permutations/SequencerPriceRouter.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { PriceRouter, Registry, ERC20, IChainlinkAggregator } from "src/modules/price-router/PriceRouter.sol";
+
+/**
+ * @title SequencerPriceRouter
+ * @notice Adds sequencer uptime feed safety checks to all PriceRouter pricing calls.
+ * @author crispymangoes
+ */
+contract SequencerPriceRouter is PriceRouter {
+    //============================== ERRORS ===============================
+
+    error SequencerPriceRouter__SequencerDown();
+    error SequencerPriceRouter__GracePeriodNotOver();
+
+    //============================== IMMUTABLES ===============================
+
+    /**
+     * @notice Address for the networks sequencer uptime feed.
+     */
+    IChainlinkAggregator internal immutable sequencerUptimeFeed;
+
+    /**
+     * @notice The amount of time that must pass from when the sequencer comes back online
+     *         to when we can continue pricing again.
+     */
+    uint256 internal immutable gracePeriod;
+
+    constructor(
+        address _sequencerUptimeFeed,
+        uint256 _gracePeriod,
+        address newOwner,
+        Registry _registry,
+        ERC20 _weth
+    ) PriceRouter(newOwner, _registry, _weth) {
+        sequencerUptimeFeed = IChainlinkAggregator(_sequencerUptimeFeed);
+        gracePeriod = _gracePeriod;
+    }
+
+    //============================== Sequencer Uptime Logic ===============================
+
+    /**
+     * @notice Layer 2 chains that use sequencers, can have the sequencer go down. If this happens we do not want
+     *         to price assets, as the datafeeds could be stale, and need to be updated.
+     */
+    function _runPreFlightCheck() internal view override {
+        (
+            ,
+            /*uint80 roundID*/ int256 answer,
+            uint256 startedAt /*uint256 updatedAt*/ /*uint80 answeredInRound*/,
+            ,
+
+        ) = sequencerUptimeFeed.latestRoundData();
+
+        // This check should make TXs from L1 to L2 revert if someone tried interacting with the cellar while the sequencer is down.
+        // Answer == 0: Sequencer is up
+        // Answer == 1: Sequencer is down
+        if (answer == 1) {
+            revert SequencerPriceRouter__SequencerDown();
+        }
+
+        // TODO better understand why this adds a grace period, and why we use startedAt instead of updatedAt.
```

Comment:
I am assuming this, but it sounds like `updatedAt` returns a nonzero value always where as `startedAt` returns a value that could be zero and is used to indicate whether or not the uptime status has changed or not. 

I'm just basing this off of [reading this part here,](https://docs.chain.link/data-feeds/l2-sequencer-feeds#:~:text=startedAt%3A%20This%20timestamp,the%20GRACE_PERIOD_TIME.) to be very sure one could dig into the `sequencerUptimeFeed` code.

Reply:
Yeah I asked chainlink for clarification, but based off that comment it sounds like it was implemented right. But we will wait to hear back from chainlink.